pkg/cmd/cli/delete_options.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/heptio/ark/pkg/client"
+	"github.com/heptio/ark/pkg/cmd/util/flag"
+	clientset "github.com/heptio/ark/pkg/generated/clientset/versioned"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// DeleteOptions contains parameters used for deleting a restore.
+type DeleteOptions struct {
+	Names            []string
+	all              bool
+	Selector         flag.LabelSelector
+	Confirm          bool
+	Client           clientset.Interface
+	Namespace        string
+	singularTypeName string
+}
+
+func NewDeleteOptions(singularTypeName string) *DeleteOptions {
+	o := &DeleteOptions{}
+	o.singularTypeName = singularTypeName
+	return o
+}
+
+// Complete fills in the correct values for all the options.
+func (o *DeleteOptions) Complete(f client.Factory, args []string) error {
+	o.Namespace = f.Namespace()
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+	o.Client = client
+	o.Names = args
+	return nil
+}
+
+// Validate validates the fields of the DeleteOptions struct.
+func (o *DeleteOptions) Validate(c *cobra.Command, f client.Factory, args []string) error {
+	if o.Client == nil {
+		return errors.New("Ark client is not set; unable to proceed")
+	}
+	var (
+		hasNames    = len(o.Names) > 0
+		hasAll      = o.all
+		hasSelector = o.Selector.LabelSelector != nil
+	)
+	if !xor(hasNames, hasAll, hasSelector) {
+		return errors.New("you must specify exactly one of: specific restore name(s), the --all flag, or the --selector flag")
```

Comment:
sorry, just spotted one more place with `restore` hardcoded that needs to be replaced with `o.singularTypeName`.  You can just amend the existing commit, no need to push a separate one.